src/abstractdataframe/join.jl

```diff
@@ -416,9 +416,13 @@ function _join(df1::AbstractDataFrame, df2::AbstractDataFrame;
     end
 
     if indicator !== nothing
-        refs = left_indicator + right_indicator
-        pool = CategoricalPool{String,UInt8}(["left_only", "right_only", "both"])
-        indicatorcol = CategoricalArray{String,1}(refs, pool)
+        left_indicator .+= right_indicator
+        pool = ["left_only", "right_only", "both"]
+        invpool = Dict{String, UInt8}("left_only" => UInt32(1),
+                                      "right_only" => UInt32(2),
+                                      "both" => UInt32(3))
```

Comment:
```suggestion
        invpool = Dict{String, UInt32}("left_only" => 1,
                                       "right_only" => 2,
                                       "both" => 3)
```

Reply:
same here

Reply:
Likewise. IMHO the presence of the type parameter is explicit enough.